sycl/test-e2e/KernelCompiler/kernel_compiler_sycl_jit.cpp

```diff
@@ -105,6 +105,34 @@ void vec_add(T* in1, T* in2, T* out){
 }
 )===";
 
+auto constexpr DeviceLibrariesSource = R"===(
+#include <sycl/sycl.hpp>
+#include <sycl/ext/intel/math.hpp>
+#include <cmath>
+
+extern "C" SYCL_EXTERNAL 
+SYCL_EXT_ONEAPI_FUNCTION_PROPERTY(sycl::ext::oneapi::experimental::single_task_kernel)
+void device_libs_kernel(float *ptr) {
+  // Extension list: llvm/lib/SYCLLowerIR/SYCLDeviceLibReqMask.cpp
+
+  // cl_intel_devicelib_assert is not available for opencl:gpu; skip testing it.
+  // Only test the fp32 variants of complex, math and imf to keep this test
+  // device-agnostic.
+  
+  // cl_intel_devicelib_math
```

Comment:
since ptr is {1, 1, 1, 1} it'd be nice to leave a comment here for each stating the expected value ( 0.8 1.4, 0, 1.4  right?)

I mention this because the test is failing on Windows with an 0xc0000409 error code, which is an assert() failure.  AFAICT, the only assert added by this PR is the  none of the new values of ptr[] are 1.0f.  

Not sure what's going on.

Reply:
In the end this was just a compilation error, due to the specific flavour of complex data type not being supported on Windows. I changed the test to use `std::complex` instead. The reason why we didn't see this in the CI logs is that the default handler for uncaught exceptions doesn't seem to print anything on Windows 🤔